app/controllers/tags_controller.rb

```diff
@@ -0,0 +1,58 @@
+class TagsController < AuthenticatedController
+  include ProjectScoped
+  include ActivityTracking
+
+  before_action :set_columns, only: :index
+  load_and_authorize_resource
+
+  def index
+    @tags = current_project.tags
+  end
+
+  def new; end
+
+  def create
+    if @tag.save
+      track_created(@tag)
+      redirect_to request.referer, notice: 'Tag created.'
+    else
+      redirect_to request.referer, alert: @tag.errors.full_messages.join('; ')
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @tag.update(tag_params)
+      track_updated(@tag)
+      redirect_to project_tags_path(current_project), notice: 'Tag updated.'
+    else
+      redirect_to project_tags_path(current_project), alert: @tag.errors.full_messages.join('; ')
+    end
+  end
+
+  def destroy
+    if @tag.destroy
+      track_destroyed(@tag)
+      redirect_to project_tags_path(current_project), alert: 'Tag deleted.'
+    else
+      redirect_to project_tags_path(current_project), alert: @tag.errors.full_messages.join('; ')
+    end
+  end
+
+  private
+
+  def tag_params
+    modified_params = params.require(:tag).permit(:display_name, :color)
```

Comment:
There's a bit of inconsistency with the names of the Tag attributes:

We have Tag/Name/Display Name on the one hand, and Color code / Color on the other.

I think we should stick with a consistent set (e.g. Name/Color).

If Display_name is a "helper" method in the model, let's not use it as an HTTP param. and if we're calling it Name in the modal, let's call it Name in the table too.

Reply:
refactored